datafusion/expr/src/signature.rs

```diff
@@ -117,7 +122,8 @@ pub enum TypeSignature {
     /// is `OneOf(vec![Any(0), VariadicAny])`.
     OneOf(Vec<TypeSignature>),
     /// Specifies Signatures for array functions
-    ArraySignature(ArrayFunctionSignature),
+    /// Boolean value specifies whether null type coercion is allowed
+    ArraySignature(ArrayFunctionSignature, bool),
```

Comment:
Could you expand this comment to explain what null type coercion is and when it would make sense *not*n to be allowed?  

I don't understand what it is used for 🤔 it seems to me like null coercion should always be allowed